commit 15f0b962f080abdc81235b05a149392b1c6c78c1
Author: q66 <q66@chimera-linux.org>
Date:   Thu Oct 15 18:00:22 2026 +0200

    wasm2c: fix SIMD_FORCE_READ on ppc64le, riscv64, loongarch64
    
    https://github.com/WebAssembly/wabt/pull/2878
    https://github.com/WebAssembly/wabt/pull/2880

diff --git a/third_party/wasm2c/src/prebuilt/wasm2c_simd_source_declarations.cc b/third_party/wasm2c/src/prebuilt/wasm2c_simd_source_declarations.cc
index 659bfdf..d88c062 100644
--- a/third_party/wasm2c/src/prebuilt/wasm2c_simd_source_declarations.cc
+++ b/third_party/wasm2c/src/prebuilt/wasm2c_simd_source_declarations.cc
@@ -16,7 +16,7 @@ R"w2c_template(#elif defined(__aarch64__)
 )w2c_template"
 R"w2c_template(#define SIMD_FORCE_READ(var) __asm__("" ::"w"(var));
 )w2c_template"
-R"w2c_template(#elif defined(__riscv)
+R"w2c_template(#elif defined(__riscv) && defined(__riscv_vector)
 )w2c_template"
 R"w2c_template(#define SIMD_FORCE_READ(var) __asm__("" ::"vr"(var));
 )w2c_template"
@@ -24,10 +24,18 @@ R"w2c_template(#elif defined(__s390x__)
 )w2c_template"
 R"w2c_template(#define SIMD_FORCE_READ(var) __asm__("" ::"d"(var));
 )w2c_template"
-R"w2c_template(#else
+R"w2c_template(#elif defined(__powerpc64__) || defined(__ppc64__)
+)w2c_template"
+R"w2c_template(#define SIMD_FORCE_READ(var) __asm__("" ::"v"(var));
+)w2c_template"
+R"w2c_template(#elif defined(__loongarch__)
 )w2c_template"
 R"w2c_template(#define SIMD_FORCE_READ(var) __asm__("" ::"f"(var));
 )w2c_template"
+R"w2c_template(#else
+)w2c_template"
+R"w2c_template(#define SIMD_FORCE_READ(var) __asm__("" ::"m"(var));
+)w2c_template"
 R"w2c_template(#endif
 )w2c_template"
 R"w2c_template(#else
